### ISSUE TITLE
Customer Group Drop-Down Not Working As Expected on Order Form

### ISSUE THREAD

**Issue by joshuaqueenava** [classyllama/ClassyLlama_AvaTax#215]
Customer POC (If needed)
Name: (male) Loren McDonald 
Email: lmcdonald@manncorp.com
Phone: (215) 830-1200
--

{*}Account Name:{*} Mann Manufacturing Services inc 
{*}Account Number(s):{*} 2000105414 
{*}Company Code(s):{*} MANNMANUFACTURINGSERVICES 
{*}AvaTax Environment:{*} Production 
{*}ERP Version:{*} Magento 2 
{*}AvaTax Integration Version:{*} avalara/avatax: 15.5.2 classyllama/module-avatax: 1.4.9 
{*}AvaTax Adapter Version:{*} 
{*}OS/Browser Version:{*} Chrome 
{*}Repro Frequency{*} When changing customer groups 

{*}Description:{*} 
When customer is changing their customer group in Magneto 2, during a manual order, they're not seeing tax recalculate when the group is changed. 

Their example is one with exemptions - they have a "Taxable" customer group, and change to an "Exempt" customer group while making the order, but tax doesn't update right away. 

This may end up being an enhancement for Classy Llama 

When creating a manual order, if the customer group is changed while in the order and then products are added to the order, the initial customer group tax shows in the totals.16:05:21 

example: I have set up a tax exempt customer group. I start an order with a customer that is in the general group... change the customer group to tax exempt and add products and there is tax generated as if it were still general.16:06:26 

when the order is submitted, the correct tax is calculated.16:06:38 

if the customer group is changed after the products are added, the tax shows up correctly in the totals area.16:07:20 

{*}Steps to Reproduce:{*} 
Create manual transaction in Magento 2 
While adding line items, update the customer group from taxable to exempt grouping 

{*}Expected Result:{*} 
Should recalculate tax after customer group change 

{*}Actual Result:{*} 
Doesn't recalculate tax until order is submitted 

{*}Attachments/Documentation:{*}

**Comment by rsisco** [comment]
@joshuaqueenava - I am unable to reproduce this problem. These are the steps I'm using:

- In Magento backend, start a new order for a customer in a taxable customer group by default
- Change the customer group on the new order to a tax exempt group
- Add a product to the order

The result is that no tax is displayed on the order. If I then change the customer group back to taxable, tax is calculated and displayed. Are you able to reproduce the issue in a vanilla Magento install?

**Comment by joshuaqueenava** [comment]
@rsisco See meeting video below.
https://avalara1-my.sharepoint.com/:v:/g/personal/joshua_queen_avalara_com/EYcAWlTK0T9Brr6UOqKZgssBhMHszCYxb7arJGmgt3Ywsw?e=c6rzob

**Comment by rsisco** [comment]
An internal ticket has been opened to look into this.

**Comment by joshuaqueenava** [comment]
@rsisco are there any updates on this ticket? Customer is considering a cancellation due to length of issue and lack of resolution.

Thank you.

**Comment by prestonchoate** [comment]
@joshuaqueenava This task is one I plan on taking care of this week. I likely won't get to start it until tomorrow or Wednesday though.

**Comment by prestonchoate** [comment]
@joshuaqueenava Good news. I was able to recreate this locally. I will be debugging through it today to try to find a resolution. I will keep you updated.

**Comment by joshuaqueenava** [comment]
Will you keep this ticket open in order to keep us updated on the Magento ticket? The customer who reported this is threatening to cancel over this, so I was hoping we can provide her updates as we get them from Magento.

**Comment by prestonchoate** [comment]
@joshuaqueenava Sure I can do that.

**Comment by joshuaqueenava** [comment]
I closed the case on my side. If you usually close these types of issues you can close this ticket.

**Comment by joshuaqueenava** [comment]
Additional info if it may help:

I just ran across a new issue on our dev site. 
Due to mas fake customer accounts being created we have decided to add using Captcha on the account creation page.
I turned that on in the dev site and when I submitted the form to test I got an error. When I tried going into the all customers page I also got the error.
 
Fatal error: Uncaught Error: Class 'ClassyLlama\AvaTax\Model\Config\Source\CrossBorderClass\Customer\ImporterOfRecord' not found in /home/loren/public_html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:93 Stack trace: #0 /home/loren/public_html/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('ClassyLlama\\Ava...', Array) #1 /home/loren/public_html/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Compiled->create('ClassyLlama\\Ava...', Array) #2 /home/loren/public_html/vendor/magento/framework/Validator/UniversalFactory.php(36): Magento\Framework\ObjectManager\ObjectManager->create('ClassyLlama\\Ava...', Array) #3 /home/loren/public_html/vendor/magento/module-eav/Model/Entity/Attribute/AbstractAttribute.php(551): Magento\Framework\Validator\UniversalFactory->create('ClassyLlama\\Ava...') #4 /home/loren/public_html/vendor/magento/module-customer/Model/AttributeMetad in /home/loren/public_html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 93
 
 
I don’t know if enabling Captcha is what started the errors or if they were there to begin with after I reinstalled yesterday. I’m doing some investigating but figured I’d see if anything about this error is known of.
 
I’ll update later with any further findings.
 
--
Thank You
Loren McDonald

**Comment by joshuaqueenava** [comment]
And this:
Joshua,
It took quite a bit of troubleshooting but I was finaly able to track the issue down to an attribute in the eav_attribute table that uses that ImporterOfRecord class as it’s source model.
Since “Seller is Importer of Record Override” is documented in the Customs, Duty & Import Tax (CDIT) section of the documentation, which wasn’t implimented until v2.x of the extension, I’m assuming this attribute has to be a left over from the v2.x install I had done earlier.
 
I had found removal isntructions in their documentation which lists some tables to remove and a column in the tax_class table, but it makes no mention of removing the attribute.
 
Anyway, once I deleted the attribute the error went away.
 
I probably won’t have any time to do testing until next week, at which time I’ll do the customer group testing you mentioned and let you know.
 
 
--
Thank You
Loren McDonald
IT/Web Development
